alvr/client_core/src/lib.rs

```diff
@@ -346,78 +334,58 @@ impl ClientCoreContext {
         }
     }
 
-    pub fn get_frame(&self) -> Option<DecodedFrame> {
-        dbg_client_core!("get_frame");
-
-        let mut decoder_source_lock = self.connection_context.decoder_source.lock();
-        let decoder_source = decoder_source_lock.as_mut()?;
-
-        let (frame_timestamp, buffer_ptr) = match decoder_source.get_frame() {
-            Ok(maybe_pair) => maybe_pair?,
-            Err(e) => {
-                error!("Error getting frame, restarting connection: {}", e);
+    /// The callback should return true if the frame was successfully submitted to the decoder
+    pub fn set_decoder_input_callback(
+        &self,
+        callback: Box<dyn FnMut(Duration, &[u8]) -> bool + Send>,
+    ) {
+        dbg_client_core!("set_decoder_input_callback");
 
-                // The connection loop observes changes on this value
-                *self.connection_context.state.write() = ConnectionState::Disconnecting;
+        *self.connection_context.decoder_callback.lock() = Some(callback);
+    }
 
-                return None;
-            }
-        };
+    pub fn report_frame_decoded(&self, timestamp: Duration) {
```

Comment:
Kinda covering multiple functions but you have observed that the client is talking the same code path as before though I assume if it didn't bad things would probably happen